hack/check-latest-images.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+
+# Copyright The Shipwright Contributors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+# Check the latest image and replace
+
+# Usage
+# ./check-latest-image.sh <FILE> <IMAGE> <IMAGE_URL>
+
+set -euo pipefail
+
+# Check the number of arguments passed and exit if less than three
+if [ $# -lt 3 ]; then
+        echo "USAGE: $0 <FILE> <IMAGE> <IMAGE_URL>"
+        exit 1
+fi
+
+# Preserve the literal value of . and /
+# sed -e 's/[]\/$*.^[]/\\&/g'
+
+# Set env for image, current version and lastest version
+IMAGE=$(printf '%s\n' "$(echo ${1})" | sed -e 's/[]\/$*.^[]/\\&/g')
+CURRENT_TAG=$(printf '%s\n' "$(grep -r -h -m 1 -o ${1}'[^ ]*' ${3} | head -n 1 | cut -d':' -f2)" | sed -e 's/[]\/$*.^[]/\\&/g')
```

Comment:
I don't really understand for what we need the `printf` here?

Adding comment

Using long flag names in `grep` to make it easier to read it.

```suggestion
# Determine the currently used tag and escape special characters.
CURRENT_TAG="$(grep --recursive --no-filename --max-count 1 -o ${1}'[^ ]*' "${3}" | head -n 1 | cut -d':' -f2)" | sed -e 's/[]\/$*.^[]/\\&/g')"
```